sklearn/preprocessing/label.py

```diff
@@ -53,6 +54,140 @@ def _check_numpy_unicode_bug(labels):
+        self.estimator = estimator
+        self.func = func
+        self.inverse_func = inverse_func
+        # we probably need to change this ones we have tags
+        self._estimator_type = estimator._estimator_type
+
+    def fit(self, X, y, sample_weight=None):
+        """Fit the model according to the given training data.
+
+        Parameters
+        ----------
+        X : {array-like, sparse matrix}, shape (n_samples, n_features)
+            Training vector, where n_samples is the number of samples and
+            n_features is the number of features.
+
+        y : array-like, shape (n_samples,)
+            Target vector relative to X.
+
+        sample_weight : array-like, shape (n_samples,) optional
+            Array of weights that are assigned to individual samples.
+            If not provided, then each sample is given unit weight.
+
+        Returns
+        -------
+        self : object
+            Returns self.
+        """
+        self.estimator_ = clone(self.estimator)
+        self.estimator_.fit(X, self.func(y), sample_weight=sample_weight)
+        return self
+
+    def predict(self, X):
+        """Predict using the base model, apply inverse_func.
+
+        Parameters
+        ----------
+        X : {array-like, sparse matrix}, shape = (n_samples, n_features)
+            Samples.
+
+        Returns
+        -------
+        y_hat : array, shape = (n_samples,)
+            Predicted values.
+        """
+        check_is_fitted(self, "estimator_")
+        return self.inverse_func(self.estimator_.predict(X))
+
+    def score(self, X, y, sample_weight=None):
+        """Computes score for regression and classification models.
+
+        For regression, r2_score is used, for classification accuracy_score.
+        See the docstring of these functions for more details.
+        The ``r2_score`` uses variance-weighted averaging in the multi-output
+        case.
+        """
+
+        check_is_fitted(self, "estimator_")
+        if is_classifier(self.estimator):
+            from ..metrics import accuracy_score
+            return accuracy_score(y, self.predict(X),
+                                  sample_weight=sample_weight)
+        elif is_regressor(self.estimator):
+            from ..metrics import r2_score
+            return r2_score(y, self.predict(X), sample_weight=sample_weight,
+                            multioutput='variance_weighted')
+        else:
+            # I'm not sure if this is too many internals for an error message
+            if not hasattr(self.estimator_, "_estimator_type"):
+                err = "estimator has declared no _estimator_type."
+            else:
+                err = "estimator has _estimator_type {}".format(
+                    self.estimator_._estimator_type)
+            raise NotImplementedError(
+                "TargetTransformer only implements a score method if "
+                "estimator is a classifier or regressor, but " + err)
+
+
```

Comment:
So you are computing the score in the "outer" domain, *not* in the "inner" domain that the estimator_ is fit.

An alternative would be: `return self.estimator_.score(X, self.func(y))` which doesn't need error checking but returns scores in the "inner" domain, where the estimator is truly fit.

I would prefer this alternative. I guess it depends on the use cases this transformer is designed for?